From 7fc3ce48266b4bc7b52d276fe2901dc052a46716 Mon Sep 17 00:00:00 2001
From: Tom Englund <tomenglund26@gmail.com>
Date: Tue, 28 Jun 2022 19:19:09 +0200
Subject: [PATCH 1/3] fix: adapt cube effect to quickview changes

commit 9095b784 changed the way current gl context is dealt with so call
effects->makeOpenGLContextCurrent() in paintScreen and the effect frame
repositioning got moved to the new quickview repositioning so set the
frame position in the constructor and let the frame auto adjust with the
text.
---
 effect/effects/cube/cube.cpp | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/effect/effects/cube/cube.cpp b/effect/effects/cube/cube.cpp
index 0e9f8c419..73719de3f 100644
--- a/effect/effects/cube/cube.cpp
+++ b/effect/effects/cube/cube.cpp
@@ -597,18 +597,20 @@ void CubeEffect::paintScreen(int mask, const QRegion& region, ScreenPaintData& d
             } else if (animationState == AnimationState::Stop) {
                 opacity = 1.0 - timeLine.value();
             }
-            QRect screenRect = effects->clientArea(ScreenArea, activeScreen, frontDesktop);
-            QRect frameRect = QRect(screenRect.width() * 0.33f + screenRect.x(),
-                                    screenRect.height() * 0.95f + screenRect.y(),
-                                    screenRect.width() * 0.34f,
-                                    QFontMetrics(desktopNameFont).height());
+
             if (!desktopNameFrame) {
-                desktopNameFrame = effects->effectFrame(EffectFrameStyled);
+                auto screen_rect = effects->clientArea(ScreenArea, activeScreen, frontDesktop);
+                auto frame_x = screen_rect.width() * 0.5f + screen_rect.x();
+                auto frame_y = screen_rect.height() * 0.95f + screen_rect.y();
+
+                desktopNameFrame = effects->effectFrame(
+                    EffectFrameStyled, false, QPoint(frame_x, frame_y), Qt::AlignCenter);
                 desktopNameFrame->setFont(desktopNameFont);
             }
-            desktopNameFrame->setGeometry(frameRect);
+
             desktopNameFrame->setText(effects->desktopName(frontDesktop));
             desktopNameFrame->render(region, opacity);
+            effects->makeOpenGLContextCurrent();
         }
     } else {
         effects->paintScreen(mask, region, data);
-- 
GitLab


From a297da953c3a174d57cfe8e0650880edc7df75b1 Mon Sep 17 00:00:00 2001
From: Tom Englund <tomenglund26@gmail.com>
Date: Tue, 28 Jun 2022 19:43:35 +0200
Subject: [PATCH 2/3] fix: adapt flipswitch effect to quickview changes

in commit 9095b784 frame repositioning/resizing got changed to use the new
class EffectQuickView and it can either be static or non static and repositions
after setting text and/or setting icon, so adapt to those changes and
set initial frame position in constructor and icon size upon creation
then let text and icon dynamicly change the frame size.
---
 effect/effects/flipswitch/flipswitch.cpp | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/effect/effects/flipswitch/flipswitch.cpp b/effect/effects/flipswitch/flipswitch.cpp
index d25e6f6f8..4024e163c 100644
--- a/effect/effects/flipswitch/flipswitch.cpp
+++ b/effect/effects/flipswitch/flipswitch.cpp
@@ -657,19 +657,19 @@ void FlipSwitchEffect::setActive(bool activate, FlipSwitchMode mode)
             break;
         }
 
-        // Setup caption frame geometry
-        QRect frameRect = QRect(m_screenArea.width() * 0.25f + m_screenArea.x(),
-                                m_screenArea.height() * 0.1f + m_screenArea.y()
-                                    - QFontMetrics(m_captionFont).height(),
-                                m_screenArea.width() * 0.5f,
-                                QFontMetrics(m_captionFont).height());
         if (!m_captionFrame) {
-            m_captionFrame = effects->effectFrame(EffectFrameStyled);
+            auto frame_x = m_screenArea.width() * 0.5f + m_screenArea.x();
+            auto frame_y = m_screenArea.height() * 0.95f + m_screenArea.y();
+
+            m_captionFrame = effects->effectFrame(
+                EffectFrameStyled, false, QPoint(frame_x, frame_y), Qt::AlignCenter);
             m_captionFrame->setFont(m_captionFont);
             m_captionFrame->enableCrossFade(true);
+            auto icon_size
+                = QSize(m_captionFrame->geometry().height(), m_captionFrame->geometry().height());
+            m_captionFrame->setIconSize(icon_size);
         }
-        m_captionFrame->setGeometry(frameRect);
-        m_captionFrame->setIconSize(QSize(frameRect.height(), frameRect.height()));
+
         updateCaption();
         effects->addRepaintFull();
     } else {
-- 
GitLab


From 2eab8c8be433a8b71ec69886b9e15ac1196730c2 Mon Sep 17 00:00:00 2001
From: Tom Englund <tomenglund26@gmail.com>
Date: Wed, 29 Jun 2022 19:00:48 +0200
Subject: [PATCH 3/3] fix: set font size to cube and flipswitch

make font sizes more consistent and use same size as desktopgrid and
presentwindows in the effect frame window.
---
 effect/effects/cube/cube.cpp             | 2 +-
 effect/effects/flipswitch/flipswitch.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/effect/effects/cube/cube.cpp b/effect/effects/cube/cube.cpp
index 73719de3f..e765aef71 100644
--- a/effect/effects/cube/cube.cpp
+++ b/effect/effects/cube/cube.cpp
@@ -99,7 +99,7 @@ CubeEffect::CubeEffect()
 {
     initConfig<CubeConfig>();
     desktopNameFont.setBold(true);
-    desktopNameFont.setPointSize(14);
+    desktopNameFont.setPointSize(12);
 
     if (effects->compositingType() == OpenGLCompositing) {
         ensureResources();
diff --git a/effect/effects/flipswitch/flipswitch.cpp b/effect/effects/flipswitch/flipswitch.cpp
index 4024e163c..7c1942239 100644
--- a/effect/effects/flipswitch/flipswitch.cpp
+++ b/effect/effects/flipswitch/flipswitch.cpp
@@ -57,7 +57,7 @@ FlipSwitchEffect::FlipSwitchEffect()
 
     // Caption frame
     m_captionFont.setBold(true);
-    m_captionFont.setPointSize(m_captionFont.pointSize() * 2);
+    m_captionFont.setPointSize(12);
 
     QAction* flipSwitchCurrentAction = new QAction(this);
     flipSwitchCurrentAction->setObjectName(QStringLiteral("FlipSwitchCurrent"));
-- 
GitLab

